docs/features/search/search-for-plugin-devs.md

```diff
@@ -0,0 +1,187 @@
+---
+id: for-plugin-devs
+title: Search for Plugin Developers
+description: How to integrate search into a Backstage plugin
+---
+
+The Backstage Search Platform was designed to give plugin developers the APIs
+and interfaces needed to offer search experiences within their plugins, while
+abstracting away (and instead empowering application integrators to choose) the
+specific underlying search technologies.
+
+On this page, you'll find concepts, guides, and recipes for how to leverage the
+Backstage Search Platform in your plugin.
```

Comment:
as you mention a couple of different packages in the text, it would be nice to have a list with links to the different ones before starting to read. 

Something like. 

`In this guide we will use <package> and <package>...`

Reply:
As I was editing this, I realized this is a guide for plugin developers, but we're talking about importing from `@backstage/plugin-search` (🙅 🙅 🙅  plugin-to-plugin dependencies 🙅 🙅 🙅 ).  So I opened up https://github.com/backstage/backstage/issues/11676.

I can add for now, but hopefully we remove it soon!

Reply:
Added a section for this!  Under the tutorials heading for experiences, since this page would also house collator stuff too. https://github.com/backstage/backstage/pull/11675/commits/9387626eeccef35f5a611aa256a5d6d02108e6c1